lib/internal/per_context/primordials.js

```diff
@@ -12,17 +12,9 @@
 // `primordials.Object` where `primordials` is a lexical variable passed
 // by the native module compiler.
 
-const ReflectApply = Reflect.apply;
-
-// This function is borrowed from the function with the same name on V8 Extras'
-// `utils` object. V8 implements Reflect.apply very efficiently in conjunction
-// with the spread syntax, such that no additional special case is needed for
-// function calls w/o arguments.
-// Refs: https://github.com/v8/v8/blob/d6ead37d265d7215cf9c5f768f279e21bd170212/src/js/prologue.js#L152-L156
 function uncurryThis(func) {
-  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
+  return Function.prototype.call.bind(func);
 }
```

Comment:
@ExE-Boss What about this:
```suggestion
const {apply} = Reflect
const {bind,call} = Function.prototype;
const uncurryThis = (func) => apply(bind, call, [func]);

Reply:
That works correctly, and so does:
```suggestion
const uncurryThis = Function.prototype.call.bind(Function.prototype.bind, Function.prototype.call);
```

Reply:
When would the name be useful? `uncurryThis` doesn't appear on error stack traces already – because it returns an arrow function, and that's this one that throws the error.